src/components/project/NewProject/NewProject.tsx

```diff
@@ -170,13 +171,19 @@ const NewProject: FC<Props> = ({
       return;
     }
 
-    if (!importURL || !active) {
+    if (
+      (projects.length == 0 && !importURL && !active) ||
+      (projects.length !== 0 && (!importURL || !active))
+    ) {
       return;
     }
+    if (projects.length === 0 && importURL) {
+      setCurrentProjectType('git');
```

Comment:
Setters should not be called from top scope of `useEffect`, because this can lead to infinite updates.

This logic is also fragile, because changes to any of `importURL, projectName, form, codeToImport` will ead to change of `currentProjectType`. It's not obvious why `projectName` effects `currentProjectType` and seems to be accidental.

Reply:
I have updated it, it won't be triggered multiple times.